SetupDataPkg/Tools/VariableList.py

```diff
@@ -1278,6 +1277,68 @@ def write_csv(schema, csv_path, full, subknobs=True):
                             knob.format.object_to_string(knob.value),
                             string_binary,
                             knob.help])
+                            
+                            
+def write_csv_detailed(schema, csv_path):
+
+    with open(csv_path, 'w', newline='') as csv_file:
+        writer = csv.writer(csv_file)
+        guid = None
+        namespace_changed = True # Used to print the guid on the first row and then only print
+                                 # another guid if it changes
+        
+        writer.writerow(['Guid', 'Knob', 'Value', 'Binary', 'Default Value', 'Valid Value Options', 'Help'])
+        # Get the knobs one at a time
+        for knob in schema.knobs:
+            # Check if the guid has changed
+            if knob.namespace == guid:
+                namespace_changed = False
+            else:
+                guid = knob.namespace
+                namespace_changed = True
+                
+            # For each subknob of the current knob, if it's a leaf, write it to the CSV file
+            for subknob in knob.subknobs:  
+                subknob_values = ""
+                
+                # Get the binary value
+                binary = subknob.format.object_to_binary(subknob.value)
+                string_binary = " ".join(map("%2.2x".__mod__, binary))
+                # Check if subknob is a leaf; skip otherwise
+                if subknob.leaf:
+                    # Get all valid values for each knob
+                    for enum in schema.enums:
+                        if enum.name in subknob.format.c_type:
+                            for entry in enum.values:
+                                subknob_values = subknob_values + ', ' + entry.name
+                            subknob_values = subknob_values.lstrip(",")
+                            break
+                        elif subknob.format.c_type == "bool":

```

Comment:
This doesn't seem like it should be within the "for enum" loop

Reply:
when you say "This doesn't seem like... " what is "this" referring to? Is this related to the next comment regarding the use of EnumFormat instead of a for loop?